gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -0,0 +1,176 @@
+  };
+
+  return body;
+};
+
+const renderBody = (body) => ({
+  facilityType: body.facilityType,
+  facilityName: body.facilityName,
+  hasBeenIssued: body.hasBeenIssued,
+  monthsOfCover: body.monthsOfCover,
+  shouldCoverStartOnSubmission: body.monthsOfCover,
+  issueDateDay: body.issueDateDay,
+  issueDateMonth: body.issueDateMonth,
+  issueDateYear: body.issueDateYear,
+  coverStartDateDay: body.coverStartDateDay,
+  coverStartDateMonth: body.coverStartDateMonth,
+  coverStartDateYear: body.coverStartDateYear,
+  coverEndDateDay: body.coverEndDateDay,
+  coverEndDateMonth: body.coverEndDateMonth,
+  coverEndDateYear: body.coverEndDateYear,
+  facilityTypeString: body.facilityTypeString,
+  dealId: body.dealId,
+  facilityId: body.facilityId,
+  status: body.status,
+  change: body.change,
+});
+
+// when changing unissued facility from unissued facility list
+const changeUnissuedFacility = async (req, res) => {
+  const { params, query } = req;
+
+  try {
+    const body = await renderChangeFacilityPartial(params, query, false);
+
+    return res.render('partials/unissued-change-about-facility.njk', renderBody(body));
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+// when changing unissued facility from application preview page
+const changeUnissuedFacilityPreview = async (req, res) => {
+  const { params, query } = req;
+
+  try {
+    const body = await renderChangeFacilityPartial(params, query, true);
+
+    return res.render('partials/unissued-change-about-facility.njk', renderBody(body));
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+/**
+ * post for changing unissued facilities to issued from unissued facilities list
+ * validates first and gets parameters from validation
+ * displays success message and redirects to unissued facilities list
+ */
+const postChangeUnissuedFacility = async (req, res) => {
+  const {
+    issueDate, coverStartDate, coverEndDate, body, facilityId, dealId,
+  } = await facilityValidation(req, res);
+
+  try {
+    await api.updateFacility(
+      facilityId,
+      {
+        name: body.facilityName,
+        shouldCoverStartOnSubmission: isTrueSet(body.shouldCoverStartOnSubmission),
+        monthsOfCover: body.monthsOfCover || null,
+        issueDate: issueDate ? format(issueDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        coverStartDate: coverStartDate ? format(coverStartDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        coverEndDate: coverEndDate ? format(coverEndDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        hasBeenIssued: true,
+        changedToIssued: true,
+        coverDateConfirmed: true,
+      },
+      (req.success = {
+        message: `${body.facilityName} is updated`,
+      }),
+      (req.url = `/gef/application-details/${dealId}/unissued-facilities`),
+    );
+    // TODO: DTFS2-5227 change redirect
+    return applicationDetails(req, res);
+  } catch (err) {
+    console.log(err);
```

Comment:
Can we please add a description before the `err` log, and change to `console.error`

Reply:
done